src/vis/vis.js

```diff
@@ -1208,7 +1208,16 @@ var Vis = View.extend({
 
     var mapView = this.mapView;
     var infowindow = null;
-    var layers = layerView.model && layerView.model.layers || [];
+    var layers = [];
+    if (layerView.model) {
+      if (layerView.model.layers) {
+        layers = layerView.model.layers;
+      } else {
+        if (layerView.model.getInfowindowData) {
+          layers = new Backbone.Collection([layerView.model]);
```

Comment:
This is probably something that each "layer view" could handle differently. For instance, if a new layer view is created and it's model has an infowindow model, it can create the corresponding InfowindowView and bind to the corresponding events (featureClick, etc). differently (also fetch attributes differently, etc.).